eos-migrate-chromium-profile

```diff
@@ -0,0 +1,151 @@
+    desktop file no longer exists.
+    """
+
+    keyfile = GLib.KeyFile()
+    try:
+        keyfile.load_from_file(
+            path, GLib.KeyFileFlags.KEEP_COMMENTS | GLib.KeyFileFlags.KEEP_TRANSLATIONS,
+        )
+    except GLib.GError as gerror:
+        if gerror.matches(GLib.file_error_quark(), GLib.FileError.NOENT):
+            return
+
+        raise
+
+    changed = False
+
+    for group in MIMEAPPS_GROUPS:
+        if not keyfile.has_group(group):
+            continue
+
+        keys, _length = keyfile.get_keys(group)
+        for key in keys:
+            values = keyfile.get_string_list(group, key)
+            try:
+                i = values.index(OLD_DESKTOP_FILE)
+            except ValueError:
+                pass
+            else:
+                values[i] = NEW_DESKTOP_FILE
+                keyfile.set_string_list(group, key, values)
+                changed = True
+
+    if changed:
+        keyfile.save_to_file(path)
+
+
+def update_desktop_shortcut(path):
+    keyfile = GLib.KeyFile()
+    print("path: " + path)
+    keyfile.load_from_file(path, GLib.KeyFileFlags.NONE)
+
+    group = "Desktop Entry"
+    if not keyfile.has_group(group):
+        return
+
+    exec_cmd = keyfile.get_string(group, "Exec")
+    if "/usr/bin/chromium-browser" not in exec_cmd:
+        return
+
+    keyfile.set_string(group, "Exec",
+                       exec_cmd.replace("/usr/bin/chromium-browser",
+                                        "/usr/bin/flatpak run org.chromium.Chromium"))
+    keyfile.set_string(group, "X-Flatpak-Part-Of", "org.chromium.Chromium")
+    keyfile.save_to_file(path)
+
+
+def update_desktop_shortcuts():
+    for filename in os.listdir(DESKTOP_SHORTCUTS_DIR):
+        if filename.endswith(".desktop"):
+            update_desktop_shortcut(os.path.join(DESKTOP_SHORTCUTS_DIR,
+                                                 filename))
+
+
+def update_old_config_references(path):
+    if not os.path.isfile(path):
+        return
+
+    for line in fileinput.input(path, inplace=True):
+        line = line.replace(OLD_CHROMIUM_CONFIG_DIR, NEW_CHROMIUM_CONFIG_DIR)
+        line = line.replace(OLD_WIDEVINE_SYSTEM_DIR, "")
+        print(line, end='')
+
+
+def main():
+    migrated_file = os.path.join(NEW_CHROMIUM_DATA_DIR, ".migrated")
+    if os.path.exists(migrated_file):
+        return
+
+    update_mimeapps_list(MIMEAPPS_LIST)
+    update_desktop_shortcuts()
+
+    if (
+        os.path.isdir(OLD_CHROMIUM_CONFIG_DIR) and
+        not os.path.isdir(NEW_CHROMIUM_CONFIG_DIR)
+    ):
+        os.makedirs(NEW_CHROMIUM_CONFIG_DIR)
+        os.rename(OLD_CHROMIUM_CONFIG_DIR, NEW_CHROMIUM_CONFIG_DIR)
+        update_old_config_references(
+            os.path.join(NEW_CHROMIUM_CONFIG_DIR,
+                         "WidevineCdm", "latest-component-updated-widevine-cdm"))
+
+    if not os.path.exists(NEW_CHROMIUM_DATA_DIR):
+        os.makedirs(NEW_CHROMIUM_DATA_DIR)
+
+    os.mknod(migrated_file)
```

Comment:
I've never seen `mknod` used to create a normal file before!

Reply:
From what I checked this was the only way to create files without opening... Would you prefer to change it?

Reply:
No, it's fine! I have always used:

```python3
with open(migrated_file, "w"):
    pass
```

to touch a file. But this (which merely needs to create, not update a timestamp) works fine too.